android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/Dataset.java

```diff
@@ -0,0 +1,302 @@
+package com.tavultesoft.kmea.data;
+
+import android.content.Context;
+import android.widget.ArrayAdapter;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.tavultesoft.kmea.data.adapters.AdapterFilter;
+import com.tavultesoft.kmea.data.adapters.ListBacked;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Dataset extends ArrayAdapter<Dataset.LanguageDataset> {
+  public abstract class LanguageFilter<Type extends LanguageCoded, Adapter extends LanguageCodedAdapter<Type>> implements AdapterFilter<Type, Adapter, String> {
+    abstract Set<Type> getSetFrom(LanguageDataset metadata);
```

Comment:
Sadly, this class was somewhat prone to iterative development across the PR chain, but this is where the class was originally introduced.

Rather than rely on massive Lists of HashMaps for everything, and then building weird frankenstein-hashmaps for the LanguagesListActivity, this class and its nested classes will provide a cleaner, better-documented, and far more UI-friendly way to manage Keyman resource data.  As it and many of its nested classes are `Adapter`-based, they're designed to pair perfectly with the new `NestedAdapter` class to facilitate reflexive/reactive UI development that provides a good Settings management UX.

The main issue for reviewing is that I missed a few details here that were needed by later work, hence fleshed out in the later PRs.